newa/cli.py

```diff
@@ -194,63 +194,125 @@ def cmd_event(ctx: CLIContext, errata_ids: list[str]) -> None:
 def cmd_jira(ctx: CLIContext, issue_config: str) -> None:
     ctx.enter_command('jira')
 
-    # this is here temporarily so we generate fake Jira issue IDs
-    jira_id_gen = itertools.count(start=1)
+    # Get control of Jira.
+    # TODO: Hardcoded project for now.
+    # It will be served from the component-config.yaml later.
+    jira = JiraHandler("BASEQESEC")
 
     for erratum_job in ctx.load_erratum_jobs('event-'):
         # read Jira issue configuration
         config = ErratumConfig.from_yaml_file(Path(os.path.expandvars(issue_config)))
 
-        # TODO: record created/existing issues. Instead of `Any`, maybe something
-        # from the Jira library would be stored. Or just a Jira ticket ID.
-        known_issues: dict[str, Any] = {}
-
-        # Iterate over issue actions. Take one, if it's not possible to finish it,
-        # put it back at the end of the queue.
+        # All issue action from the configuration.
         issue_actions = config.issues[:]
 
+        # Issues related to the curent respin.
+        new_issues: dict[str, Issue] = {}
+
+        # Issue actions for issues related to previous respin(n).
+        previous_issues_actions: list[tuple[str, OnRespinAction]] = []
+
+        # First, go through issue actions to see what issues from the previous
+        # respin(s) need to be processed.
+        for action in issue_actions:
+
+            # Find existing issues in Jira related to erratum_job and action.id.
+            #
+            # We re looking for Jira issues such that:
+            #
+            # 1) summary matches rendered_summary (e.g. Errata respin 0) and
+            # 2) description contains erratum_job.erratum (e.g. Erratum(130704) and
+            # 3) description contains erratum_job.erratum.release (e.g. Release(RHEL-8.8.0.Z.EUS))
+            query = f""" summary ~ '{render_template(action.summary, ERRATUM=erratum_job)}' and
+                         description ~ '{erratum_job.erratum!r}' and
+                         description ~ '{erratum_job.erratum.release!r}' """
+            for jira_issue in jira.search_open_issue(query):
+                desc = jira_issue.fields.description
+
+                # We found relevant issue in Jira, it is either related to the
+                # current respin (new) or not (old).
+                #
+                # 1) If it is, we encountered error (either none or all issues must exist).
+                if isinstance(desc, str) and desc.find(repr(erratum_job.erratum.builds)):
+                    raise Exception(
```

Comment:
I believe it would be perfectly valid to run `newa` command multiple times for a single artifact and re-use already existing Jira issue. This could be e.g. because a user wants to re-run previously aborted tests.

Reply:
Alright, this requires change. Also, thanks for the suggestion for the loop below. I guess I misunderstood the previous remark about creating new issues before the old are closed - I was under impression that we want to create all new one before handling all old ones which is most likely not what you meant (based on the loop suggestion). I must admit I don't like the idea of complete split (what I did in the current version) very much. So I am glad I can get back from it.

Reply:
The last version should reflect your proposal with some rather cosmetic differences and some part of the code from the previous version. It might not be semantically 100% bullet-proof yet because I am just about to run it for the first time now. So please expect some additional fixes but I hope we have an agreement on the overall structure now :).

Reply:
Fixed in the current version.